acceptance/framework/consul/cli_cluster.go

```diff
@@ -102,92 +105,94 @@ func NewCLICluster(
 	}
 }
 
-func (h *CLICluster) Create(t *testing.T) {
+// Create uses the `consul-k8s install` command to create a Consul cluster. The command itself will fail if there are
+// prior installations of Consul in the cluster so it is sufficient to run the install command without a pre-check.
+func (c *CLICluster) Create(t *testing.T) {
 	t.Helper()
 
 	// Make sure we delete the cluster if we receive an interrupt signal and
 	// register cleanup so that we delete the cluster when test finishes.
-	helpers.Cleanup(t, h.noCleanupOnFailure, func() {
-		h.Destroy(t)
+	helpers.Cleanup(t, c.noCleanupOnFailure, func() {
+		c.Destroy(t)
 	})
 
-	// The install command itself will fail if there are prior installations, so it's sufficient to just run the install
-	// command.
+	// Set the args for running the install command.
 	args := []string{"install"}
-	kubeconfig := h.kubeConfig
-	if kubeconfig != "" {
-		args = append(args, "-kubeconfig", kubeconfig)
+	args = c.setKube(args)
+
+	for k, v := range c.values {
+		args = append(args, "-set", fmt.Sprintf("%s=%s", k, v))
 	}
-	kubecontext := h.kubeContext
-	if kubecontext != "" {
-		args = append(args, "-context", kubecontext)
+
+	// Match the timeout for the helm tests.
+	args = append(args, "-timeout", "15m")
+	args = append(args, "-auto-approve")
+
+	out, err := c.runCLI(args)
+	if err != nil {
+		c.logger.Logf(t, "error running command `consul-k8s %s`: %s", strings.Join(args, " "), err.Error())
+		c.logger.Logf(t, "command stdout: %s", string(out))
 	}
+	require.NoError(t, err)
 
-	for k, v := range h.values {
-		args = append(args, "-set", fmt.Sprintf("%s=%s", k, v))
+	k8s.WaitForAllPodsToBeReady(t, c.kubernetesClient, consulNS, fmt.Sprintf("release=%s", c.releaseName))
+}
+
+// Upgrade uses the `consul-k8s upgrade` command to upgrade a Consul cluster.
+func (c *CLICluster) Upgrade(t *testing.T, helmValues map[string]string) {
+	t.Helper()
+
+	k8s.WritePodsDebugInfoIfFailed(t, c.kubectlOptions, c.debugDirectory, "release="+c.releaseName)
```

Comment:
If there are failed pods at this point, it'd be from the install before an upgrade. Should we potentially abort the upgrade at this point and return out of this function and fail the test?

Reply:
Great thought! I added the check below this to abort if the test is failing. Let me know what you think.

Reply:
Yeah that looks good!